cmd/juju/waitfor/application.go

```diff
@@ -38,13 +38,31 @@ func newApplicationCommand() cmd.Command {
 }
 
 const applicationCommandDoc = `
-Waits for an application to reach a specified state.
-arguments:
-name
-   application name identifier
-options:
---query (= 'life=="alive" && status=="active"')
-   query represents the sought state of the specified application
+The wait-for application command waits for the application to reach a goal
+state. The goal state can be defined programmatically using the query DSL
+(domain specific language). The default query for an application just waits
+for the application to be created and active.
+
+The wait-for command is an optimized alternative to the status command for 
+determining programmatically if a goal state has been reached. The wait-for
+command streams delta changes from the underlying database, unlike the status
+command which performs a full query of the database.
+
+The application query DSL can be used to programmatically define the goal state
+for machines and units within the scope of the application. This can
+be achieved by using lambda expressions to iterate over the machines and units
+associated with the application. Multiple expressions can be combined to define 
+a complex goal state.
+`
+const applicationCommandExamples = `
+    Waits for 4 units to be present.
```

Comment:
This looks like too much indentation. The main text shouldn't be indented -- just the code blocks.

Reply:
It looks really weird if we don't indent.